trino/sqlalchemy/dialect.py

```diff
@@ -0,0 +1,314 @@
+    # Data Type
+    supports_native_enum = False
+    supports_native_boolean = True
+    supports_native_decimal = True
+
+    # Column options
+    supports_sequences = False
+    supports_comments = True
+    inline_comments = True
+    supports_default_values = False
+
+    # DDL
+    supports_alter = True
+
+    # DML
+    supports_empty_insert = False
+    supports_multivalues_insert = True
+    postfetch_lastrowid = False
+
+    @classmethod
+    def dbapi(cls):
+        """
+        ref: https://www.python.org/dev/peps/pep-0249/#module-interface
+        """
+        return trino_dbapi
+
+    def create_connect_args(self, url: URL) -> Tuple[List[Any], Dict[str, Any]]:
+        args, kwargs = super(TrinoDialect, self).create_connect_args(url)  # type: List[Any], Dict[str, Any]
+
+        db_parts = kwargs.pop('database', 'system').split('/')
+        if len(db_parts) == 1:
+            kwargs['catalog'] = db_parts[0]
+        elif len(db_parts) == 2:
+            kwargs['catalog'] = db_parts[0]
+            kwargs['schema'] = db_parts[1]
+        else:
+            raise ValueError(f'Unexpected database format {url.database}')
+
+        username = kwargs.pop('username', 'anonymous')
+        kwargs['user'] = username
+
+        password = kwargs.pop('password', None)
+        if password:
+            kwargs['http_scheme'] = 'https'
+            kwargs['auth'] = BasicAuthentication(username, password)
+
+        return args, kwargs
+
+    def get_columns(self, connection: Connection,
+                    table_name: str, schema: str = None, **kw) -> List[Dict[str, Any]]:
+        if not self.has_table(connection, table_name, schema):
+            raise exc.NoSuchTableError(f'schema={schema}, table={table_name}')
+        return self._get_columns(connection, table_name, schema, **kw)
+
+    def _get_columns(self, connection: Connection,
+                     table_name: str, schema: str = None, **kw) -> List[Dict[str, Any]]:
+        schema = schema or self._get_default_schema_name(connection)
+        query = dedent('''
+            SELECT
+                "column_name",
+                "column_default",
+                UPPER("is_nullable"),
+                "data_type"
+            FROM "information_schema"."columns"
+            WHERE "table_schema" = :schema AND "table_name" = :table
+            ORDER BY "ordinal_position" ASC
+        ''').strip()
+        res = connection.execute(sql.text(query), schema=schema, table=table_name)
+        columns = []
+        for record in res:
+            column = dict(
+                name=record.column_name,
+                type=datatype.parse_sqltype(record.data_type),
+                nullable=record.is_nullable == 'YES',
+                default=record.column_default,
+            )
+            columns.append(column)
+        return columns
+
+    def get_pk_constraint(self, connection: Connection,
+                          table_name: str, schema: str = None, **kw) -> Dict[str, Any]:
```

Comment:
Any specific way we are splitting arguments across lines? 

Maybe we should introduce an autoformatter like `black` here?

(No changes requested)